src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -59,9 +63,93 @@ constexpr int8_t kDefaultDeadBand                 = 25; // 2.5C is the default
 #define FEATURE_MAP_SCH 0x08
 #define FEATURE_MAP_SB 0x10
 #define FEATURE_MAP_AUTO 0x20
+#define FEATURE_MAP_LTNE 0x40
 
 #define FEATURE_MAP_DEFAULT FEATURE_MAP_HEAT | FEATURE_MAP_COOL | FEATURE_MAP_AUTO
 
+class ThermostatAttrAccess : public AttributeAccessInterface
+{
+public:
+    ThermostatAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), Thermostat::Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+};
+
+ThermostatAttrAccess gThermostatAttrAccess;
+
+CHIP_ERROR ThermostatAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == Thermostat::Id);
+
+    bool LTNESupported = false;
+    uint32_t OurFeatureMap;
+
+    if (FeatureMap::Get(aPath.mEndpointId, &OurFeatureMap) == EMBER_ZCL_STATUS_SUCCESS &&
+        OurFeatureMap & 1 << 6) // Bit 6 is LTNE supported
+        LTNESupported = true;
+
+    switch (aPath.mAttributeId)
+    {
+    case LocalTemperature::Id: {
+        if (LTNESupported)
+            return aEncoder.EncodeNull();
+        break;
+    }
+    case RemoteSensing::Id: {
+        if (LTNESupported)
+        {
+            uint8_t valueRemoteSensing;
+            VerifyOrReturnValue(RemoteSensing::Get(aPath.mEndpointId, &valueRemoteSensing) == EMBER_ZCL_STATUS_SUCCESS,
+                                CHIP_NO_ERROR);
+            valueRemoteSensing &= 0xFE; // clear bit 1 (LocalTemperature RemoteSensing bit)
+            return aEncoder.Encode(valueRemoteSensing);
+        }
+        break;
+    }
+    default:
+        break;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ThermostatAttrAccess::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    VerifyOrDie(aPath.mClusterId == Thermostat::Id);
+
+    bool LTNESupported = false;
+    uint32_t OurFeatureMap;
+
+    if (FeatureMap::Get(aPath.mEndpointId, &OurFeatureMap) == EMBER_ZCL_STATUS_SUCCESS &&
+        OurFeatureMap & 1 << 6) // Bit 6 is LTNE supported
+        LTNESupported = true;
+
+    switch (aPath.mAttributeId)
+    {
+    case RemoteSensing::Id: {
+        if (LTNESupported)
+        {
+            uint8_t valueRemoteSensing;
+            ReturnErrorOnFailure(aDecoder.Decode(valueRemoteSensing));
+            if (valueRemoteSensing & 0x01) // If setting bit 1 (LocalTemperature RemoteSensing bit)
+            {
+                return StatusIB(imcode::ConstraintError).ToChipError();
+            }
+            else
+            {
+                RemoteSensing::Set(aPath.mEndpointId, valueRemoteSensing);
```

Comment:
See above for how an EmberAfStatus can get turned into a CHIP_ERROR.  Just do that unconditionally here and return; EMBER_ZCL_STATUS_SUCCESS will turn into CHIP_NO_ERROR.